### PR TITLE
WFP-1143

### PR DESCRIPTION
**Pull request by markreesmoj**
Work in progress - a couple of problems
- I found adding reactive at the same time to be too large a change. You can see how far I got in the previous commit
- the stock response does not match what the API produces according to Swagger:  the risk is a String apparently. Found it hard to deserialise this (also not sure why it should be a string rather than an object)

Feel free to have a go at fixing either or both.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/controller/UnallocatedCasesController.kt [hunk 1]
```diff
@@ -91,7 +92,7 @@ class UnallocatedCasesController(
         ?: throw EntityNotFoundException("Unallocated case Not Found for $crn")
     )
 
-  @Operation(summary = "Retrieve unallocated case risks by crn")
+  @Operation(summary = "Retrieve unallocated case risks by crn DEPRECATED")
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/controller/UnallocatedCasesController.kt:95]: There used to be a `Deprecated` annotation for Swagger..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/domain/UnallocatedCaseRisks.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ data class UnallocatedCaseRisks @JsonCreator constructor(
   val tier: String,
   val activeRegistrations: List<UnallocatedCaseRegistration>,
   val inactiveRegistrations: List<UnallocatedCaseRegistration>,
-  val rosh: UnallocatedCaseRosh?,
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/domain/UnallocatedCaseRisks.kt:20]: How are you intending to swap over to the new version? Without backwards compatibility there will have to be a bit of down time - albeit a few mins probably.

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/domain/UnallocatedCaseRisks.kt:20]: The new implementation is a different endpoint. The old one is still there and unchanged. It is unwieldy and probably not very clear from this change, but basically I have renamed all the existing classes with the suffix of Deprecated. We can then switch the UI to the new endpoint and delete the Deprecated classes and controller endpoint. That will save coming up with a name like risksNew
On the endpoint itself, I was thinking to push another change after the UI has moved over, making the new endpoint available on `/risks` as well as `/risksNew`, switching the UI to that endpoint and then deleting `/risksNew` just for tidiness. It was either that or put a version somewhere in the URL or header, neither of which is particularly fancied.
It occurs to me now that I could have added a new Rosh object with a different name to the existing endpoint.

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/domain/UnallocatedCaseRisks.kt:20]: Taken a new approach for no downtime

**Pull request merged**